src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8413,617 +9036,195 @@ void SketchObject::rebuildExternalGeometry(bool defining, bool addIntersection)
+
         // Skip frozen geometries
         bool frozen = false;
         bool sync = false;
-        bool intersection = addIntersection && (i+1 == (int)Objects.size());
         for(auto id : externalGeoRefMap[key]) {
             auto it = externalGeoMap.find(id);
             if(it != externalGeoMap.end()) {
                 auto egf = ExternalGeometryFacade::getFacade(ExternalGeo[it->second]);
-                if(egf->testFlag(ExternalGeometryExtension::Frozen))
+                if(egf->testFlag(ExternalGeometryExtension::Frozen)) {
                     frozen = true;
-                if(egf->testFlag(ExternalGeometryExtension::Sync))
+                }
+                if (egf->testFlag(ExternalGeometryExtension::Sync)) {
                     sync = true;
+                }
             }
         }
         if(frozen && !sync) {
             refSet.insert(std::move(key));
             continue;
         }
-        if(!Obj || !Obj->getNameInDocument())
+        if (!Obj || !Obj->getNameInDocument()) {
             continue;
+        }
 
         std::vector<std::unique_ptr<Part::Geometry> > geos;
 
+        auto importVertex = [&](const TopoDS_Shape& refSubShape) {
+            gp_Pnt P = BRep_Tool::Pnt(TopoDS::Vertex(refSubShape));
+            GeomAPI_ProjectPointOnSurf proj(P, gPlane);
+            P = proj.NearestPoint();
+            Base::Vector3d p(P.X(), P.Y(), P.Z());
+            invPlm.multVec(p, p);
+
+            Part::GeomPoint* point = new Part::GeomPoint(p);
+            GeometryFacade::setConstruction(point, true);
+            geos.emplace_back(point);
+        };
+
         try {
-        TopoDS_Shape refSubShape;
+            TopoDS_Shape refSubShape;
 
-        if (Obj->isDerivedFrom<Part::Datum>()) {
-            const Part::Datum* datum = static_cast<const Part::Datum*>(Obj);
-            refSubShape = datum->getShape();
-        }
-        else if (Obj->isDerivedFrom<Part::Feature>()) {
-            try {
-                const Part::Feature* refObj = static_cast<const Part::Feature*>(Obj);
-                const Part::TopoShape& refShape = refObj->Shape.getShape();
-                refSubShape = refShape.getSubShape(SubElement.c_str());
+            if (Obj->isDerivedFrom<Part::Datum>()) {
+                auto* datum = static_cast<const Part::Datum*>(Obj);
+                refSubShape = datum->getShape();
             }
-            catch (Standard_Failure& e) {
-                throw Base::CADKernelError(e.GetMessageString());
+            else if (Obj->isDerivedFrom<Part::Feature>()) {
+                try {
+                    auto* refObj = static_cast<const Part::Feature*>(Obj);
+                    const Part::TopoShape& refShape = refObj->Shape.getShape();
+                    refSubShape = refShape.getSubShape(SubElement.c_str());
+                }
+                catch (Standard_Failure& e) {
+                    throw Base::CADKernelError(e.GetMessageString());
+                }

```

Comment:
There's a `try` within a `try`, and the `Base::CADKernelError` that it throws seems to only trigger the `catch (Base::Exception ...)` rather than the `catch (Standard_Failure)`. Maybe we can remove this safely.